src/server/host/Endpoints/Events/PatchPlayerEventRegistrationsEndpoint.cs

```diff
@@ -0,0 +1,190 @@
+using System.ComponentModel.DataAnnotations;
+using FastEndpoints;
+using FluentValidation;
+using MaSch.Core.Extensions;
+using Microsoft.AspNetCore.Server.HttpSys;
+using Microsoft.EntityFrameworkCore;
+using MinigolfFriday.Data;
+using MinigolfFriday.Data.Entities;
+using MinigolfFriday.Domain.Models;
+using MinigolfFriday.Domain.Models.RealtimeEvents;
+using MinigolfFriday.Host.Common;
+using MinigolfFriday.Host.Services;
+
+namespace MinigolfFriday.Host.Endpoints.Events;
+
+/// <param name="EventId">The id of the event to change registration.</param>
+/// <param name="TimeslotId">The registration to change.</param>
+/// <param name="IsRegistered">The registration state to change the timeslot to.</param>
+/// <param name="UserId">The userId to patch. Only available as Admin.</param>
+public record PatchPlayerEventRegistrationsRequest(
+    [property: Required] string EventId,
+    [property: Required] string TimeslotId,
+    [property: Required] bool IsRegistered,
+    [property: Required] string UserId
+);
+
+public class PatchPlayerEventRegistrationsRequestValidator
+    : Validator<PatchPlayerEventRegistrationsRequest>
+{
+    public PatchPlayerEventRegistrationsRequestValidator(IIdService idService)
+    {
+        RuleFor(x => x.EventId).NotEmpty().ValidSqid(idService.Event);
+        RuleFor(x => x.TimeslotId).NotEmpty().ValidSqid(idService.EventTimeslot);
+    }
+}
+
+public class PatchPlayerEventRegistrationsEndpoint(
+    DatabaseContext databaseContext,
+    IRealtimeEventsService realtimeEventsService,
+    IIdService idService,
+    IJwtService jwtService
+) : Endpoint<PatchPlayerEventRegistrationsRequest>
+{
+    public override void Configure()
+    {
+        Patch("{eventId}/registrations");
+        Group<EventsGroup>();
+        this.ProducesErrors(
+            EndpointErrors.UserIdNotInClaims,
+            EndpointErrors.EventNotFound,
+            EndpointErrors.EventRegistrationElapsed,
+            EndpointErrors.EventAlreadyStarted,
+            EndpointErrors.UserNotFound
+        );
+    }
+
+    public override async Task HandleAsync(
+        PatchPlayerEventRegistrationsRequest req,
+        CancellationToken ct
+    )
+    {
+        if (!jwtService.TryGetUserId(User, out var userId))
+        {
+            Logger.LogWarning(EndpointErrors.UserIdNotInClaims);
+            await this.SendErrorAsync(EndpointErrors.UserIdNotInClaims, ct);
+            return;
+        }
+        if (!jwtService.HasRole(User, Role.Admin))
```

Comment:
Instead of this, configure the endpoint, so it can only be called by users with the admin role. This can be done in the `Configure` method by calling the `Roles` method in there. This will lead to a 401 when the user has not the correct role.

Reply:
actually the 403 <forbidden> will be returned now - if this is the recommended way of doing it